src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBufferProvider.ts

```diff
@@ -0,0 +1,123 @@
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { DisposableStore } from 'vs/base/common/lifecycle';
+import { IModelService } from 'vs/editor/common/services/model';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { TerminalCapability, ITerminalCommand } from 'vs/platform/terminal/common/capabilities/capabilities';
+import { ICurrentPartialCommand } from 'vs/platform/terminal/common/capabilities/commandDetectionCapability';
+import { TerminalSettingId } from 'vs/platform/terminal/common/terminal';
+import { AccessibilityVerbositySettingId } from 'vs/workbench/contrib/accessibility/browser/accessibilityConfiguration';
+import { AccessibleViewType, IAccessibleContentProvider, IAccessibleViewOptions, IAccessibleViewService, IAccessibleViewSymbol } from 'vs/workbench/contrib/accessibility/browser/accessibleView';
+import { IXtermTerminal, ITerminalInstance, ITerminalService } from 'vs/workbench/contrib/terminal/browser/terminal';
+import { BufferContentTracker } from 'vs/workbench/contrib/terminalContrib/accessibility/browser/bufferContentTracker';
+import type { Terminal } from 'xterm';
+import { Event } from 'vs/base/common/event';
+
+export class TerminalAccessibleBufferProvider extends DisposableStore implements IAccessibleContentProvider {
+	options: IAccessibleViewOptions = { type: AccessibleViewType.View };
+	verbositySettingKey = AccessibilityVerbositySettingId.Terminal;
+	private _xterm: IXtermTerminal & { raw: Terminal } | undefined;
+	constructor(
+		private readonly _instance: Pick<ITerminalInstance, 'onDidRunText' | 'focus' | 'shellType' | 'capabilities' | 'onDidRequestFocus' | 'resource'>,
+		private _bufferTracker: BufferContentTracker,
+		@IModelService _modelService: IModelService,
+		@IConfigurationService _configurationService: IConfigurationService,
+		@IContextKeyService _contextKeyService: IContextKeyService,
+		@ITerminalService _terminalService: ITerminalService,
+		@IConfigurationService configurationService: IConfigurationService,
+		@IAccessibleViewService private readonly _accessibleViewService: IAccessibleViewService
+	) {
+		super();
+		this.add(_instance.onDidRunText(() => {
+			const focusAfterRun = configurationService.getValue(TerminalSettingId.FocusAfterRun);
+			if (focusAfterRun === 'terminal') {
+				_instance.focus(true);
+			} else if (focusAfterRun === 'accessible-buffer') {
+				_accessibleViewService.show(this);
+			}
+		}));
+		this.registerListeners();
+	}
+
+	onClose() {
+		this._instance.focus();
+	}
+	registerListeners(): void {
+		if (!this._xterm) {
+			return;
+		}
+		this._xterm.raw.onWriteParsed(async () => {
+			if (this._xterm!.raw.buffer.active.baseY === 0) {
+				this.provideContent();
```

Comment:
This updates the buffer tracker and also constructs and returns a string it doesn't need